tools/linter/adapters/test_device_bias_linter.py

```diff
@@ -0,0 +1,155 @@
+            for subnode in ast.walk(node):
+                if isinstance(subnode, ast.keyword):
+                    if subnode.arg == "device":
+                        val = subnode.value
+                        if isinstance(val, ast.Constant) and DEVICE_BIAS in val.value:
+                            # detect: device="cuda"
+                            self.record(subnode, f"{msg_prefix} device='{val.value}'")
+                        elif isinstance(val, ast.Call):
+                            # detect: device=torch.device("cuda")
+                            if (
+                                isinstance(val.func, ast.Attribute)
+                                and val.func.attr == "device"
+                                and len(val.args) > 0
+                                and isinstance(val.args[0], ast.Constant)
+                                and DEVICE_BIAS in val.args[0].value
+                            ):
+                                self.record(
+                                    val,
+                                    f"{msg_prefix} torch.device('{val.args[0].value}')",
+                                )
+                if isinstance(subnode, ast.Call) and isinstance(
+                    subnode.func, ast.Attribute
+                ):
+                    method_name = subnode.func.attr
+                    if method_name == DEVICE_BIAS:
+                        # detect: .cuda()
+                        self.record(subnode, f"{msg_prefix} .{DEVICE_BIAS}() call")
+                    elif method_name == "to":
+                        # detect: to("cuda")
+                        if subnode.args:
+                            arg = subnode.args[0]
+                            if (
+                                isinstance(arg, ast.Constant)
+                                and DEVICE_BIAS in arg.value
+                            ):
+                                self.record(
+                                    subnode, f"{msg_prefix} .to('{arg.value}') call"
+                                )
+
+        self.generic_visit(node)
+
+    def record(self, node: ast.AST, message: str) -> None:
+        self.lint_messages.append(
+            LintMessage(
+                path=self.filename,
+                line=getattr(node, "lineno", None),
+                char=None,
+                code=LINTER_CODE,
+                severity=LintSeverity.ERROR,
+                name="[device-bias]",
+                original=None,
+                replacement=None,
+                description=message,
+            )
+        )
+
+
+def check_file(filename: str) -> list[LintMessage]:
+    with open(filename) as f:
+        source = f.read()
+        tree = ast.parse(source, filename=filename)
+        checker = DeviceBiasVisitor(filename)
+        checker.visit(tree)
+
+    return checker.lint_messages
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="test device bias code",
```

Comment:
`Detect Python functions decorated with [require_gpu] may potentially break support for other GPU devices.`

Reply:
> `Detect Python functions decorated with [require_gpu] may potentially break support for other GPU devices.`

Then it should be decorated with requires_cuda instead of requires_gpu.

Reply:
nit: my comment is to refine the description.